Tests/ResilientDecodingTests/ResilientDictionaryTests.swift

```diff
@@ -0,0 +1,159 @@
+    XCTAssertEqual(mock.$optionalResilientDictionary.errors.count, 0)
+    #endif
+  }
+
+  func testDecodesNullValue() throws {
+    let mock = try decodeMock(ResilientDictionaryWrapper.self, """
+      {
+        "resilientDictionary": null,
+        "optionalResilientDictionary": null,
+      }
+      """)
+    XCTAssertEqual(mock.resilientDictionary, [:])
+    XCTAssertNil(mock.optionalResilientDictionary)
+    #if DEBUG
+    XCTAssert(mock.$resilientDictionary.outcome.is(.valueWasNil))
+    XCTAssert(mock.$optionalResilientDictionary.outcome.is(.valueWasNil))
+    XCTAssertEqual(mock.$resilientDictionary.errors.count, 0)
+    XCTAssertEqual(mock.$optionalResilientDictionary.errors.count, 0)
+    #endif
+  }
+
+  func testResilientlyDecodesIncorrectType() throws {
+    let mock = try decodeMock(ResilientDictionaryWrapper.self, """
+      {
+        "resilientDictionary": 1,
+        "optionalResilientDictionary": 1,
+      }
+      """,
+      expectedErrorCount: 2)
+    XCTAssertEqual(mock.resilientDictionary, [:])
+    XCTAssertEqual(mock.optionalResilientDictionary, [:])
+    #if DEBUG
+    XCTAssert(mock.$resilientDictionary.outcome.is(.recoveredFromError(wasReported: true)))
+    XCTAssertEqual(mock.$resilientDictionary.errors.count, 1)
+    XCTAssert(mock.$optionalResilientDictionary.outcome.is(.recoveredFromError(wasReported: true)))
+    XCTAssertEqual(mock.$optionalResilientDictionary.errors.count, 1)
+    #endif
+  }
+
+  func testResilientlyDecodesArrayWithInvalidElements() throws {
+    let mock = try decodeMock(ResilientDictionaryWrapper.self, """
+      {
+        "resilientDictionary": {
+          "1": 1,
+          "2": "2",
+          "3": 3,
+          "4": "4",
+          "5": 5,
+        },
+        "optionalResilientDictionary": {
+          "1": "1",
+          "2": 2,
+          "3": "3",
+          "4": "4",
+          "5": 5,
+        },
+      }
+      """,
+      expectedErrorCount: 5)
+    XCTAssertEqual(mock.resilientDictionary, ["1": 1, "3": 3, "5": 5])
+    XCTAssertEqual(mock.optionalResilientDictionary, ["2": 2, "5": 5])
+    #if DEBUG
+    XCTAssert(mock.$resilientDictionary.outcome.is(.recoveredFromError(wasReported: false)))
+    XCTAssertEqual(mock.$resilientDictionary.errors.count, 2)
+    XCTAssertEqual(mock.$resilientDictionary.results.mapValues { try? $0.get() }, ["1": 1, "2": nil, "3": 3, "4": nil, "5": 5])
+    XCTAssert(mock.$optionalResilientDictionary.outcome.is(.recoveredFromError(wasReported: false)))
+    XCTAssertEqual(mock.$optionalResilientDictionary.errors.count, 3)
+    XCTAssertEqual(mock.$optionalResilientDictionary.results.mapValues { try? $0.get() }, ["1": nil, "2": 2, "3": nil, "4": nil, "5": 5])
+    #endif
+  }
+
+  /**
+   This tests internal functionality which decodes a dictionary of results to make sure the coding paths aren't impacted by https://bugs.swift.org/browse/SR-6294
+   */
+  func testDictionaryOfResults() throws {
```

Comment:
It would be helpful if this test name said what it's testing.

Reply:
I'd also recommending wrapping this whole test in `#if DEBUG`. No point in having it even run in Release IMHO since it gives a false sense of security.

Reply:
I haven't tried, but I think this will break `swift test --generate-linuxmain`, since I'm not sure it can handle conditional tests.

Reply:
And the default is always run both sets of tests